app/controllers/saml_controller.rb

```diff
@@ -1,10 +1,15 @@
 # frozen_string_literal: true
 
 class SamlController < ApplicationController
-  skip_before_action :authenticate, only: [:metadata]
+  skip_before_action :authenticate, only: [:metadata, :metadata_v2]
 
   def metadata
     meta = OneLogin::RubySaml::Metadata.new
-    render xml: meta.generate(saml_settings), content_type: 'application/xml'
+    render xml: meta.generate(SAML::SettingsService.saml_settings), content_type: 'application/xml'
+  end
+
+  def metadata_v2
```

Comment:
let's fold this method into the new sessions_controller class, I don't see a strong reason for a separate controller just for the metadata, and that will clear. the way to eventually remove some of. the saml_settings stuff from application_controller where it doesn't really belong.